core/src/main/scala/iliad/gl/cachedgl.scala

```diff
@@ -139,23 +139,122 @@ object CachedGL {
         } yield ()
     }
 
+  def load(t: Texture.Constructor,
+           d: Option[Texture.Data]): DSL[Texture.Loaded] =
+    Cached.get(t).freekF[CachedGL] flatMap {
+      case Some(tl) => Free.pure(tl)
+      case None =>
+        for {
+          tl <- Load(t, d).freekF[CachedGL]
+          _ <- Cached.put(tl).freekF[CachedGL]
+        } yield tl
+    }
+
+  def renderbuffer(r: Renderbuffer.Constructor): DSL[Renderbuffer.Loaded] =
+    Cached.get(r).freekF[CachedGL] flatMap {
+      case Some(rl) => Free.pure(rl)
+      case None =>
+        for {
+          rl <- Load(r).freekF[CachedGL]
+          _ <- Cached.put(rl).freekF[CachedGL]
+        } yield rl
+    }
+
+  private def attachment(a: Framebuffer.AttachmentConstructor)
+    : DSL[String Xor Framebuffer.AttachmentLoaded] = a match {
+    case t: Texture.Constructor =>
+      Cached
+        .ensure(Cached.get(t), s"Texture missing $t")
+        .freekF[CachedGL]
+        .map(identity)
+    case r: Renderbuffer.Constructor =>
+      Cached
+        .ensure(Cached.get(r), s"Renderbuffer missing $r")
+        .freekF[CachedGL]
+        .map(identity)
+  }
+
+  private def attachments(f: Framebuffer.Constructor): DSL[
+      String Xor List[(FramebufferAttachment, Framebuffer.AttachmentLoaded)]] =
+    f.attachments.map {
+      case (c, a) => XorT(attachment(a)).map(al => c -> al).value
+    }.sequence.map(xors => xors.sequence)
+
+  def load(f: Framebuffer.Constructor): DSL[String Xor Unit] =
+    Cached.get(f).freekF[CachedGL] flatMap {
+      case Some(fl) => Free.pure(().right)
+      case None =>
+        (for {
+          as <- XorT(attachments(f))
+          fl <- xort(Load(f, as).freekF[CachedGL])
+          _ <- xort(Cached.put(fl).freekF[CachedGL])
+        } yield ()).value
+    }
+
   def clear(bitMask: ChannelBitMask): DSL[Unit] =
     Draw.clear(bitMask).freekF[CachedGL]
 
   private def doIfNot(f: Current.DSL[Boolean])(g: DSL[Unit]): DSL[Unit] =
     f.freekF[CachedGL] flatMap (b => if (b) Free.pure(()) else g)
```

Comment:
Let `g` be an `Eval` or lazy. Can we also call this something different? I think `ensure` is the standard ish name.